pyuvdata/uvdata/uvdata.py

```diff
@@ -589,6 +697,1062 @@ def _set_flex_spw(self):
+            UVData attribute object_name).
+        object_type : str
+            Type of object to be added. Must be one of the following:
+                sidereal - Most commonly used, described typically by RA/Dec
+                ephem - Describes "moving" objects, e.g., solar system bodies
+                driftscan - Fixed az/el position (assumes zenith if no args supplied)
+                unphased - Multi-obj equivalent of phase_type == "drift"
+        object_lon : float or ndarray
+            Value of the longitudinal coordinate (e.g., RA, Az, l) of the object.
+            No default, not used for unphased objects. Expected to be a float for
+            sidereal and driftscan objects, and an ndarray of floats of shape (Npts,)
+            for ephem objects.
+        object_lat : float or ndarray
+            Value of the latitudinal coordinate (e.g., Dec, El, b) of the object.
+            No default, not used for unphased objects. Expected to be a float for
+            sidereal and driftscan objects, and an ndarray of floats of shape (Npts,)
+            for ephem objects.
+        coord_frame : str
+            Coordinate frame that object_lon and object_lat are given in. Only used
+            for sidereal and ephem targets. Can be any of the several supported frames
+            in astropy (a limited list: fk4, fk5, icrs, gcrs, cirs, galactic).
+        coord_epoch : str or float
+            Epoch of the coordinates, only used when coord_frame = fk4 or fk5. Given
+            in unites of fractional years, either as a float or as a string with the
+            epoch abbreviation (e.g, Julian epoch 2000.0 would be J2000.0).
+        object_pm_ra : float
+            Proper motion in RA, in units of mas/year. Only used for sidereal objects.
+        object_pm_dec : float
+            Proper motion in Dec, in units of mas/year. Only used for sidereal objects.
+        object_dist : float or ndarray of float
+            Distance of the source, in units of pc. Only used for sidereal and ephem
+            objects. Expected to be a float for sidereal and driftscan objects, and an
+            ndarray of floats of shape (Npts,) for ephem objects.
+        object_vrad : float or ndarray of float
+            Radial velocity of the source, in units of km/s. Only used for sidereal and
+            ephem objects. Expected to be a float for sidereal and driftscan objects,
+            and an ndarray of floats of shape (Npts,) for ephem objects.
+        object_coord_time : ndarray of float
+            Time (in Julian day) for each set of coordinates in an ephemeris. Only used
+            for ephem objects, shape (Npts,).
+        object_src : str
+            Optional string describing the source of the information provided. Used
+            primarily in UVData to denote when an ephemeris has been supplied by the
+            JPL-Horizons system, or user-supplied. Default is 'user'.
```

Comment:
The docstring is actually a tad out of date  -- presently, it can take on three different values: 'user' for when a user has manually added it, 'jphl' for when it's been generated via astroquery and JPL-Horizons, and 'file' for when the data were read in from from a file.  The second case is obviously ephem types only, but the first and third case apply to all of them. More generally, I kinda wanted to have a place to stuff "reference" information for all sources, although in the way that its named, I can see that's not immediately clear. I'll see if I can cook up a better name/description.

Reply:
Fixed and renamed to `info_source` in 867287f05d6f65c1529d7ac97545c6327b184332.